source/s2n/s2n_tls_channel_handler.c

```diff
@@ -697,17 +697,9 @@ static void s_tls_key_operation_complete_common(
 
     AWS_ASSERT((error_code != 0) ^ (output != NULL)); /* error_code XOR output must be set */
 
-    if (operation == NULL) {
-        AWS_LOGF_ERROR(AWS_LS_IO_TLS, "Operation: Operation is null and therefore cannot be set to complete!");
-        return;
-    }
-
-    // Ensure this can only be called once and exactly once. If called again, log an error.
+    /* Ensure this can only be called once and exactly once. */
     size_t complete_count = aws_atomic_fetch_add(&operation->complete_count, 1);
-    if (complete_count != 0) {
-        AWS_LOGF_ERROR(AWS_LS_IO_TLS, "Operation id=%p: Operation is already complete!", (void *)operation);
-        return;
-    }
+    AWS_FATAL_ASSERT(complete_count == 0);
```

Comment:
trivial: writing fatal asserts like this will give users a bit of a clue what's going on
`AWS_FATAL_ASSERT(complete_count == 0 && "TLS key operation marked complete multiple times");`